app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/splash/SplashScreen.kt

```diff
@@ -1,58 +1,206 @@
+                    )
+                )
+            }
         }
     }
-
     Box(
         modifier = Modifier.fillMaxSize(),
         contentAlignment = Alignment.Center
     ) {
         Image(
-            painter = painterResource(id = R.drawable.splash_bg),
+            painter = painterResource(id = if (showLoginComponents) R.drawable.ic_overlay else R.drawable.splash_bg),
             contentDescription = stringResource(id = R.string.splash_background_content_description),
             modifier = Modifier.matchParentSize(),
             contentScale = ContentScale.Crop
         )
         Image(
             painter = painterResource(id = R.drawable.ic_logo_white),
             contentDescription = stringResource(id = R.string.splash_logo_content_description),
-            modifier = Modifier.size(201.0.dp, 48.0.dp)
+            modifier = Modifier
+                .size(201.0.dp, 48.0.dp)
+                .offset {
+                    IntOffset(
+                        x = logoOffset.value.x.toInt(),
+                        y = logoOffset.value.y.toInt()
+                    )
+                }
         )
+        AnimatedVisibility(visible = showLoginComponents, enter = fadeIn()) {
+            LoginComponents()
+        }
     }
     viewModel.startTimerToNavigateToLogin()
 }
 
+@Suppress("LongMethod")
+@Composable
+fun LoginComponents() {
+    Column(verticalArrangement = Arrangement.spacedBy(20.dp)) {
+        var enableLoginButton by remember { mutableStateOf(false) }
+        var emailState by remember { mutableStateOf("") }
+        var passwordState by remember { mutableStateOf("") }
+
+        EmailTextField(
+            emailState = emailState,
+            onValueChange = {
+                emailState = it
+                enableLoginButton = emailState.isNotEmpty() && passwordState.isNotEmpty()
+            }
+        )
+        PasswordTextField(
+            passwordState = passwordState,
+            onValueChange = {
+                passwordState = it
+                enableLoginButton = emailState.isNotEmpty() && passwordState.isNotEmpty()
+            }
+        )
+        LoginButton(loginButtonState = enableLoginButton)
+    }
+}
+
+@Composable
+fun EmailTextField(emailState: String, onValueChange: (String) -> Unit) {
+    TextField(
+        value = emailState,
+        onValueChange = {
+            onValueChange(it)
+        },
+        shape = RoundedCornerShape(12.dp),
+        singleLine = true,
+        label = { Text(LocalContext.current.getString(R.string.login_email)) },
+        colors = textFieldColor(),
+        modifier = Modifier.loginTextFieldModifier(),
+        trailingIcon = {
+            if (emailState.isNotEmpty()) {
+                IconButton(onClick = { onValueChange("") }) {
+                    Icon(
+                        imageVector = Icons.Outlined.Close,
+                        contentDescription = null
+                    )
+                }
+            }
+        }
```

Comment:
Nice addition, but since it's not included in the [Figma design](https://www.figma.com/file/GjRPOjDyZ6f4EDL3wKarRK/Challenge---Mobile-App?node-id=34%3A1450) we can probably remove it 😉